src/v/kafka/server/group_manager.cc

```diff
@@ -106,15 +109,20 @@ ss::future<> group_manager::stop() {
     }
 
     return _gate.close().then([this] {
-        /**
-         * cancel all pending group opeartions
-         */
-        for (auto& [_, group] : _groups) {
-            group->shutdown();
-        }
+        shutdown_groups();
+        _partitions.clear();
     });
 }
 
+void group_manager::shutdown_groups() {
+    /**
+     * cancel all pending group opeartions
+     */
+    for (auto& [_, group] : _groups) {
+        group->shutdown();
+    }
+}
```

Comment:
it seems like this was factored out of `::stop` so it could be reused. but it doesn't appear to be reused. is that an oversight?